internal/api/option/auth.go

```diff
@@ -243,6 +255,43 @@ func (a *authInterceptor) WrapStreamingHandler(
 		return next(ctx, conn)
 	}
 }
+func (a *authInterceptor) listServiceAccounts(
+	ctx context.Context,
+	username string,
+	groups []string,
+) (map[string][]string, error) {
+	queries := []libClient.MatchingFields{
+		{
+			kubeclient.ServiceAccountsBySubjectIndexField: username,
+		},
+	}
+	for _, group := range groups {
+		queries = append(queries, libClient.MatchingFields{
+			kubeclient.ServiceAccountsByGroupIndexField: group,
+		})
+	}
+	accounts := make(map[string]map[string]struct{})
```

Comment:
Could we save ourselves from more looping later for every authz by building this collection differently here at authn time?

```
accounts := make(map[string]map[types.NamespacedName]struct{})
```